spyder/plugins/explorer/widgets/fileassociations.py

```diff
@@ -0,0 +1,525 @@
+                    item = self.list.item(row)
+                    if app == item.text() and fpath == item.fpath:
+                        break
+                else:
+                    item = None
+        elif os.name == 'nt':
+            if fpath is None:
+                basedir = 'C:\\'
+                filters = _('Applications (*.exe *.bat *.com)')
+                title = _('Select application')
+                fpath, __ = getopenfilename(self, title, basedir, filters)
+
+            if fpath and fpath.endswith(('.exe', '.bat', '.com')):
+                app = os.path.basename(fpath).capitalize().rsplit('.')[0]
+                for row in range(self.list.count()):
+                    item = self.list.item(row)
+                    if app == item.text() and fpath == item.fpath:
+                        break
+                else:
+                    item = None
+        else:
+            if fpath is None:
+                basedir = '/'
+                filters = _('Applications (*.desktop)')
+                title = _('Select application')
+                fpath, __ = getopenfilename(self, title, basedir, filters)
+
+            if fpath and fpath.endswith(('.desktop')):
+                entry_data = parse_linux_desktop_entry(fpath)
+                app = entry_data['name']
+                for row in range(self.list.count()):
+                    item = self.list.item(row)
+                    if app == item.text() and fpath == item.fpath:
+                        break
+                else:
+                    item = None
+
+        if fpath:
+            if item:
+                self.list.setCurrentItem(item)
+            elif app:
+                icon = get_application_icon(fpath)
+                item = QListWidgetItem(icon, app)
+                item.fpath = fpath
+                self.list.addItem(item)
+                self.list.setCurrentItem(item)
+
+        self.list.setFocus()
+
+    def filter(self, text):
+        """Filter the list of applications based on text."""
+        text = self.edit_filter.text().lower().strip()
+        for row in range(self.list.count()):
+            item = self.list.item(row)
+            item.setHidden(text not in item.text().lower())
+
+    def set_extension(self, extension):
+        """Set the extension on the label of the dialog."""
+        self.label.setText(_('Choose the application for files of type ')
+                           + extension)
+
+    @property
+    def application_path(self):
+        """Return the selected application path to executable."""
+        return self.list.currentItem().fpath
+
+    @property
+    def application_name(self):
+        """Return the selected application name."""
+        return self.list.currentItem().text()
+
+
+class FileAssociationsWidget(QWidget):
+    """Widget to add applications association to file extensions."""
+    _REGEX = r'(?:(?:\*{1,1}|\w+)\.\w+)(?:,(?:\*{1,1}|\w+)\.\w+){0,20}'
```

Comment:
What does this regex do? Could you give it a more descriptive name?

Reply:
Will update